rpc/common/moc_common_networkcommon.proto

```diff
@@ -51,4 +51,10 @@ message Route {
 	string nextHop = 1;
 	string destinationPrefix = 2;
 	uint32 metric = 3;
+}
+
+message ResourceReference {
+	string Name = 1;
+	string Location = 2;
+	string Group = 3;
```

Comment:
Why do we have `Group` here if NSGs are scoped to location?

Reply:
This is meant to be a very general thing - we'll need to use something like this in the future for backreferences across NC so I included both location and group so that the reference can use whichever it's parented to. Group isn't used for NSG

Reply:
If it's supposed to be a generic thing, let's put it common_common.proto. If it's specifically meant for nsg, let's remove group and name it appropriately.

Reply:
It's generic in the case of networking - we want to reuse this for other references in SDN, like slb references

Reply:
I see, in that case it makes sense in network common but let's rename it and fix the fields. Let's remove group.

Reply:
Or we can call it `GroupSubResource` and place it in common_common.proto

Reply:
i am not sure if we need to introduce groupsubresource, locationsubresource etc. just having one is good enough. depending on the scope the fields will be populated. this is a pattern in azure too where they use a single id string which may can be the full subscription/resourcegroup/resource etc or a subset based on the context its used.

Reply:
Again, we are trying to influence rpc based on the SDK models. It may make sense in SDK but it's better to keep things specific in protobufs so that we don't have to write if else conditions in the cloud agent code to figure out if Group resource or Location resource.

Reply:
we dont need if else, scope of a resource is well defined.

Reply:
Right. Since it's well defined, IMO it makes sense to bring that protobufs so that it brings clarity to the code.

We can keep them generic in the SDK.